streams/src/test/java/org/apache/kafka/streams/StreamsBuilderTest.java

```diff
@@ -59,6 +59,7 @@
     public void shouldAllowJoinUnmaterializedFilteredKTable() {
         final KTable<Bytes, String> filteredKTable = builder.<Bytes, String>table("table-topic").filter(MockPredicate.<Bytes, String>allGoodPredicate());
         builder.<Bytes, String>stream("stream-topic").join(filteredKTable, MockValueJoiner.TOSTRING_JOINER);
+        builder.build();
```

Comment:
Should we just do `final ProcessorTopology topology = builder.build();` and remove the line below? Same elsewhere.

Reply:
Actually, we still need both The `StreamsBuilder.build()` call returns a `Topology` while the `InternalTopologyBuilder.build()` returns a `ProcessoryTopology`.   


While the method names seem to be a bit overloaded, the terminology and methods pre-date this PR, so maybe we can do a follow-up PR to look at renaming and clarifying things some.

Reply:
Ack, thanks for clarifying.